sympy/solvers/diophantine.py

```diff
@@ -2262,8 +2261,6 @@ def descent(A, B):
         return (1, 0, 1)
     if A == 1:
         return (1, 1, 0)
-    if B == -1:
-        return (None, None, None)
```

Comment:
It is already handled by the routine. But the case of `A = B = -1` leads to infinite recursion without that; I added it back with a test.